### PR TITLE
Replace `admin_default` js autocomplete

### PR DESCRIPTION
**Pull request by shmshd**
Replace `jquery-ui` autocomplete with `tom-select` selector.

### CODE REVIEW

**Comment by shmshd** [comment]
autocomplete

**Comment by evrifaessa** [comment]
Thanks for the PR, it looks really nice. 🙂

But I feel like we first should discuss about importing Axios or not.

Some time ago, we've started migrating away from jQuery's AJAX to the native fetch() API.

Do we really need another dependency, Axios here?

**Comment by shmshd** [comment]
> Some time ago, we've started migrating away from jQuery's AJAX to the native fetch() API.
> 
> Do we really need another dependency, Axios here?

The reason to choose `axios` is its browser support due to the `XMLHttpRequest` usage under the hood. The difference between both is almost 2% as per [caniuse.com](https://caniuse.com/).

**Comment by jaapmarcus** [comment]
> Do we really need another dependency, Axios here?

Main difference is browsers older than 7 years. Should be care about it?
Screenshot 2023-03-01 at 11 44 03

**Comment by BelleNottelling** [comment]
No, we shouldn't be worried about outdated web browsers. Maybe for the client area, but certainly not the admin panel

**Comment by evrifaessa** [comment]
Old browsers shouldn't be much of a concern. I don't think it's enough of a reason to stop using the native fetch API in favor of Axios.

If it got merged, we would be using 3 different ways to do requests, AJAX, Fetch and Axios. Fetch by itself is pretty sufficient.

**Comment by John-S4** [comment]
Agree with all of the above, especially about not needing to support ancient browsers (including on the client area).

File: src/modules/Invoice/html_admin/mod_invoice_index.html.twig [hunk 1]
```diff
@@ -282,37 +296,7 @@
...
-            select: function(event, ui) {
-                $(".js-autocomplete").val(ui.item.label);
-                $(".client_id").val(ui.item.value);
-
-                return false;
-            }
-        });
-    });
-
+    console.log();
```
Comment (jaapmarcus) [inline src/modules/Invoice/html_admin/mod_invoice_index.html.twig:299]: Can be removed

**Review by jaapmarcus — CHANGES_REQUESTED** [review]

**Comment by shmshd** [comment]
Now client ID is shown in the selection result to easily identify the client from duplicate names.

autocomplete-d

**Comment by jaapmarcus** [comment]
Can we also include the "Company" name?

**Comment by BelleNottelling** [comment]
> Can we also include the "Company" name?

Not all of them will have that set

**Comment by shmshd** [comment]
> Can we also include the "Company" name?

I did not want to change the API response. API provides only ID and Name.

**Comment by jaapmarcus** [comment]
> > Can we also include the "Company" name?
> 
> Not all of them will have that set

I know but for example: User Y from  Company X calls and past worker who is in the contacts left few months ago I still need to search them them en the current method is going trough all the full list. 

If present we should include them in the api response and search functions

**Review by BelleNottelling — APPROVED** [review]

**Review by evrifaessa — APPROVED** [review]
Well done. Thanks for the PR 🙂

**Review by jaapmarcus — APPROVED** [review]

**Pull request merged**